homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -393,6 +409,81 @@ def end(self) -> None:
             )
         )
 
+    async def prepare_wake_word_detection(self) -> None:
+        """Prepare wake-word-detection."""
+        # Need to add to pipeline store
+        engine = wake_word.async_default_engine(self.hass)
+        assert engine is not None, "No wake word engine"
+
+        wake_provider = wake_word.async_get_wake_word_detection_entity(
+            self.hass, engine
+        )
+        if wake_provider is None:
+            raise WakeWordDetectionError(
+                code="wake-provider-missing",
+                message=f"No wake-word-detection provider for: {engine}",
+            )
+
+        self.wake_engine = engine
+        self.wake_provider = wake_provider
+
+    async def wake_word_detection(
+        self,
+        stream: AsyncIterable[bytes],
+        audio_buffer: list[bytes],
+    ) -> wake_word.DetectionResult | None:
+        """Run wake-word-detection portion of pipeline. Returns detection result."""
+        self.process_event(
+            PipelineEvent(
+                PipelineEventType.WAKE_START,
+                {
+                    "engine": self.wake_engine,
+                },
```

Comment:
Should this event include the audio metadata like STT-start ?

Reply:
And if we only allow a single metadata type, do we check that in validate?

Reply:
I've included a hard-coded metadata like stt but without the `language` field. I don't think we need to validate at this point because wake/stt share the same audio stream, and stt will convert to the right format.